charts/argo-cd/templates/argocd-configs/argocd-rbac-cm.yaml

```diff
@@ -1,17 +1,17 @@
-{{-  if .Values.server.rbacConfigCreate }}
+{{- if (coalesce .Values.server.rbacConfigCreate .Values.configs.rbac.create) }}
 apiVersion: v1
 kind: ConfigMap
 metadata:
   name: argocd-rbac-cm
   labels:
     {{- include "argo-cd.labels" (dict "context" . "component" .Values.server.name "name" "rbac-cm") | nindent 4 }}
-  {{- with .Values.server.rbacConfigAnnotations }}
+  {{- with (coalesce .Values.server.rbacConfigAnnotations .Values.configs.rbac.annotations) }}
   annotations:
     {{- range $key, $value := . }}
     {{ $key }}: {{ $value | quote }}
     {{- end }}
   {{- end }}
-{{- with .Values.server.rbacConfig }}
+{{- with (coalesce .Values.server.rbacConfig (omit .Values.configs.rbac "create" "annotations")) }}
```

Comment:
Coalesce makes this backward compatible change - Users can upgrade and then migrate to new section.